zyte_spider_templates/spiders/ecommerce.py

```diff
@@ -92,7 +93,16 @@ class EcommerceSpider(Args[EcommerceSpiderParams], BaseSpider):
     @classmethod
     def from_crawler(cls, crawler: Crawler, *args, **kwargs) -> scrapy.Spider:
         spider = super(EcommerceSpider, cls).from_crawler(crawler, *args, **kwargs)
-        spider.allowed_domains = [get_domain(spider.args.url)]
+        seed_url = spider.args.seed_url
+        if seed_url:
+            response = requests.get(seed_url)
+            urls = [url.strip() for url in response.text.split("\n")]
+            urls = [url for url in urls if url]
```

Comment:
Would you mind moving parsing logic to a function, for easier testing & documentation?